File: apzkr-pzpi-21-5-vyshniak-viktoriia/Task1-Server/BLL/Infrastructure/Models/Institution/InstitutionModel.cs
using BLL.Infrastructure.Models.Arduino;
using BLL.Infrastructure.Models.Facility;
using Common.Enums;

namespace BLL.Infrastructure.Models.Institution;
public class InstitutionModel
{
    public int InstitutionId { get; set; }

    public int OwnerId { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public string PhoneNumber { get; set; }

    public string Address { get; set; }

    public double Latitude { get; set; }

    public double Longitude { get; set; }

    public InstitutionType InstitutionType { get; set; } = InstitutionType.Cafe;

    public string Logo { get; set; }

    public RatingModel Rating { get; set; }

    public Region Region { get; set; }

    public string WebsiteUrl { get; set; }

    public IEnumerable<FacilityListItem> Facilities { get; set; }

    public RFIDSettingsModel RFIDSettings { get; set; }
}

File: apzkr-pzpi-21-5-vyshniak-viktoriia/Task3-WebClient/src/app/core/models/institution/institution-model.ts
import { InstitutionType } from "../../enums/institution-type";
import { Region } from "../../enums/region";
import { FacilityListItem } from "../facility/facility-list-item";
import { RatingModel } from "./rating-model";

export interface InstitutionModel {
	institutionId: number;
	ownerId: number;
	name: string;
	description: string;
	phoneNumber: string;
	address: string;
	latitude: number;
	longitude: number;
	institutionType: InstitutionType; 
	logo: string;
	rating: RatingModel;
	region: Region;
	regionName: string;
	institutionTypeName: string;
	websiteUrl: string;
	facilities: FacilityListItem[];
}
